dashboard/lib/services/level_files.rb

```diff
@@ -0,0 +1,75 @@
+module Services
+  module LevelFiles
+    # Creates or updates the corresponding .level file for the given level
+    #
+    # @param [Level]
+    # @return [String] path to the .level file
+    def self.write_custom_level_file(level)
+      # Don't update the file system unless we both expect to have a file for
+      # this level and are ready for that file to be updated.
+      return unless Policies::LevelFiles.write_to_file?(level) && level.published
+
+      file_path = Policies::LevelFiles.level_file_path(level.name)
+      File.write(file_path, level.to_xml)
+      file_path
+    end
+
+    # Deletes the corresponding .level file for a given level
+    #
+    # @param [Level]
+    def self.delete_custom_level_file(level)
+      # Don't update the file system unless we expect to have a file for this level.
+      return unless Policies::LevelFiles.write_to_file?(level)
+
+      # TODO: not sure why this isn't using level_file_path
+      file_path = Dir.glob(Rails.root.join("config/scripts/**/#{level.name}.level")).first
```

Comment:
I've left this implemented the way it used to be for now, because I don't want to introduce functional changes in a reorganization PR, even if they do seem safe and simple. I plan to address this TODO when I implement the changes to start putting `.level` files in subdirectories